verifyFiles.py

```diff
@@ -0,0 +1,151 @@
+            md5_sum=file_hash(os.path.join(rootdir, filename),hasher)
+	elif os.path.isdir(os.path.join(rootdir, filename)):
+            md5_sum=directory_hash(hasher,os.path.join(rootdir, filename))
+	return md5_sum
+
+#Method file_hash is used for generating md5 checksum of a file Input: file name and hasher
+def file_hash(fileName,hasher):
+    fileContent=open(fileName)
+    while True:
+        readBuffer=fileContent.read(2**20)
+        if len(readBuffer)==0: 
+	   break
+        hasher.update(readBuffer)
+    fileContent.close()
+    return hasher.hexdigest()
+
+#Method directory_hash recursively traverses in case a directory is found, sorts the contents and creates checksum on the entire read content.
+def directory_hash(hasher, dirPath):
+    if os.path.isdir(dirPath):
+        for entry in sorted(os.listdir(dirPath)):
+            directory_hash(hasher, os.path.join(dirPath, entry))
+    elif os.path.isfile(dirPath):
+        hasher.update(file_hash(dirPath,hasher))
+    else: pass
+    return hasher.hexdigest()
+
+#Method generate_common_files_list will create a list containing the common elements from the different dictionaries corresponding to the conditions in which it was created
+def generate_common_files_list(study_folder_details_dict_list,fileWithDirectoryDetails):
+	common_files_list=[]
+	keys_list=read_contents_from_file(fileWithDirectoryDetails)
+	index=0
+	common_set=set()
+	reference_dict=study_folder_details_dict_list[index]
+	reference_set=set(reference_dict[keys_list[0]].keys())
+	for item in study_folder_details_dict_list:
+              dictionary=item[keys_list[index]]
+	      keys_from_each_dictionary=set(dictionary.keys())
+	      if not common_set:
+	         common_set=reference_set & keys_from_each_dictionary
+	      else:
+		 common_set=common_set & keys_from_each_dictionary
+	      index+=1
+	#dict(set.intersection(*(set(d.iteritems()) for d in dicts)))
+	common_files_list=list(common_set)
+    	return common_files_list
+
+#Method generate_missing_files will create a list containing the files that are not common to all the subject folders. It takes study_folder_details_dict_list,fileWithDirectoryDetails and common_files_list as the input parameters.
+def generate_missing_files_list(study_folder_details_dict_list,fileWithDirectoryDetails,common_files_list):
+	missing_files_list=[]
+	keys_list=read_contents_from_file(fileWithDirectoryDetails)
+	index=0;
+	keys_from_all_files=set()
+	for item in study_folder_details_dict_list:
+	     dictionary=item[keys_list[index]]
+	     keys_from_individual_files=set(dictionary.keys())
+	     keys_from_all_files=keys_from_all_files | keys_from_individual_files
+	     index+=1
+	missing_files_list=list(keys_from_all_files - set(common_files_list))
+	return missing_files_list
+	     
+
+def main():
+        parser=argparse.ArgumentParser()
+        parser.add_argument('file_in', help='Input the text file containing the path to the subject folders')
```

Comment:
Explain what the content of this file should be. Explain what is a subject folder. I guess you should also refer to different conditions here.